curvefs/src/mds/topology/topology.cpp

```diff
@@ -1129,129 +1133,118 @@ TopoStatusCode TopologyImpl::UpdatePartitionTxIds(
     return TopoStatusCode::TOPO_STORGE_FAIL;
 }
 
-// choose random
-int TopologyImpl::GetOneRandomNumber(int start, int end) const {
-    static std::random_device rd;
-    static std::mt19937 gen(rd());
-    std::uniform_int_distribution<> dis(start, end);
-    return dis(gen);
-}
-
-TopoStatusCode TopologyImpl::ChooseSinglePoolRandom(
-    PoolIdType *out, const std::set<PoolIdType> &unavailablePools) const {
-    poolMutex_.RDLock();
-    std::unordered_map<PoolIdType, Pool> tmpPoolMap = poolMap_;
-    poolMutex_.Unlock();
-
-    // erase unavailable pools
-    for (const auto &pool : unavailablePools) {
-        auto it = tmpPoolMap.find(pool);
-        if (it != tmpPoolMap.end()) {
-            tmpPoolMap.erase(it);
+TopoStatusCode TopologyImpl::ChooseRecoveredMetaServer(
```

Comment:
The algorithm should be similar to ChooseAvailableMetaServers, considering disk capacity

Reply:
> The algorithm should be similar to ChooseAvailableMetaServers, considering disk capacity

fixed